docs/dev/adr/0004-group-protection-for-the-service-accounts.md

```diff
@@ -0,0 +1,68 @@
+
+## Potential Solutions
+
+1. Rely on adding the `groups` claim to the Service Accounts via the Package CR configuration and the `Hardcoded claim` Protocol Mapper
+
+This option relies on pursuing the "Expose a way to add additional Claims and Protocol Mappers to the allows list for UDSClientPolicyPermissionsExecutor [#697](https://github.com/defenseunicorns/uds-identity-config/issues/697)" ticket. The next step is to add the `oidc-hardcoded-claim-mapper` (that supports hardcoding Claims into the token) to the allow list of Protocol Mappers. Once this is done, the user leverages the `spec.sso.[*].protocolMappers` and `spec.sso.[*].protocolMappers.Config` parts of the Package CR to add hardcoded `groups` claim to the Service Account tokens.
+
+Advantages:
+
+- No additional changes required
+
+Disadvantages:
+
+- Requires two-step configuration (adding Protocol Mapper to allow list and then configuring it on the Package CR)
+
+2. Introduce the `/UDS Core/Admin/Service Account` and `/UDS Core/Auditor/Service Account` groups, and automatically add the appropriate `groups` claim to Service Account tokens
+
+This approach relies on extending the existing Group/Permission model in UDS Core down to Service Accounts. With this approach every Service Account that belongs `/UDS Core/Auditor/Service Account` is automatically member of the `/UDS Core/Auditor` group which makes the Package CR configuration simpler. An additional Protocol Mapper would be automatically added to every Keycloak Client that has Service Account enabled, which would add the `groups` claim with proper values based on the Service Account group membership.
+
+Advantages:
+
+- Fine-level control over Service Account permissions
+
+Disadvantages:
+
+- Extends the existing Group/Permission model of the UDS Core Platform, which requires re-evaluating the Threat Model
+
+3. Extend the existing Package CR `sso` spec with a protection mechanism based on custom claims
+
+The implementation assumes extending the Package CR `sso` spec with a new mechanism custom claims checks, such as `azp` (that contains the name of the Service Account) or `preferred_username` (usually `service-account-<client-id>`). The UDS Core would adjust the `RequestAuthentication` to inspect the `@request.auth.claims.groups` (as described in the [Istio documentation](https://istio.io/latest/docs/tasks/security/authentication/jwt-route/#configuring-ingress-routing-based-on-jwt-claims)) and grant request access based on the presence of specific custom claims.
```

Comment:
Is it accurate that this would only handle our custom plugin group protection, but doesn't help for end user apps that might specifically require the group claim?

Reply:
That is true - I added a Disadvantage about this. Please check if it looks good and resolve this thread.